### ISSUE TITLE
Mismatch of natted IPs on different subnets than source NAT

### ISSUE THREAD

**Issue by nvazquez** [apache/cloudstack#3539]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
Networking
~~~

##### CLOUDSTACK VERSION
~~~
4.11.2
~~~

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### SUMMARY
VMs with public IPs on different subnet than the source NAT IP get the source NAT IP instead of the natted IP 

##### STEPS TO REPRODUCE
Create isolated network
Assign NAT IPs to VM's with public IPs on different subnet than source NAT subnet, 
[Restart VR with cleanup]

##### EXPECTED RESULTS
~~~
VM public IP = natted IP 
~~~

##### ACTUAL RESULTS
~~~
VM public IP = the source NAT IP
~~~

**Comment by rhtyd** [comment]
I think this was fixed, unless the regression came back again. Can you test latest master @nvazquez @borisstoyanov @andrijapanicsb ?

**Comment by rhtyd** [comment]
As per https://github.com/apache/cloudstack/pull/2476 this was fixed in 4.11.3.0

**Comment by nvazquez** [comment]
Hi @rhtyd, can you confirm the fix is not in 4.11.2 but on 4.11.3? Asking as the fix says milestone 4.11.1

**Comment by weizhouapache** [comment]
@nvazquez this issue should have been fixed in 4.11.1, however you reported the issue existed in 4.11.2
